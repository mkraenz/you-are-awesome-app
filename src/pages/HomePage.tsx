import React, { Component } from "react";
import { View } from "react-native";
import { Button, Header, Icon } from "react-native-elements";
import { HeaderTitle } from "react-navigation-stack";
import { connect } from "react-redux";
import { Navigation } from "../Navigation";
import { IPostContent } from "../redux/IPost";
import { IReduxState } from "../redux/IReduxState";
import OfflineNotice from "./components/OfflineNotice";
import RefreshableHomePageMainView from "./components/RefreshableHomePageMainView";
import { INavigationProps } from "./INavigationProps";
import { styles } from "./Styles";

interface Props extends INavigationProps {
    post: IPostContent;
    connectedToInternet: boolean;
}

class HomePage extends Component<Props> {
    public render() {
        return (
            <View style={styles.container}>
                <Header
                    centerComponent={
                        <HeaderTitle style={styles.header}>Home</HeaderTitle>
                    }
                    rightComponent={
                        <Icon
                            name="settings"
                            onPress={() =>
                                this.props.navigation.navigate(
                                    Navigation.Settings
                                )
                            }
                            color={styles.header.color}
                        />
                    }
                ></Header>
                {this.shouldBeLoading() && <OfflineNotice />}

                <RefreshableHomePageMainView />
                <Button
                    title="Share awesomeness"
                    onPress={() =>
                        this.props.navigation.navigate(Navigation.Contribute)
                    }
                />
            </View>
        );
    }

    private shouldBeLoading() {
        return (
            !this.props.connectedToInternet &&
            this.props.post.text === "Loading..." // slightly hacky
        );
    }
}

const mapStateToProps = (state: IReduxState): Omit<Props, "navigation"> => ({
    post: state.app.currentPost,
    connectedToInternet: state.netInfo.connected,
});

export default connect(mapStateToProps)(HomePage);
