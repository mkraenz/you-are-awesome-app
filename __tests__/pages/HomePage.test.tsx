import React from "react";
import "react-native";
import { Provider } from "react-redux";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import createMockStore from "redux-mock-store";
import HomePage from "../../src/pages/HomePage";

it("renders correctly", () => {
    const store = createMockStore([])({
        app: {
            currentPost: {
                id: 1,
                author: "my-author",
                text: "my-text",
                country: "my-country",
            },
        },
    });

    const tree = renderer
        .create(
            <Provider store={store}>
                <HomePage />
            </Provider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

// TODO test interactivity
