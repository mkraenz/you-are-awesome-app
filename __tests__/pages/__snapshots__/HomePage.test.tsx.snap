// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#89cff0",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
  >
    <Image
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 64,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="headerContainer"
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        accessibilityTraits="header"
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontSize": 17,
            "fontWeight": "600",
            "marginHorizontal": 16,
          }
        }
      >
        Home
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#89cff0",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "flex": 1,
        }
      }
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
        />
      }
    >
      <RCTRefreshControl />
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#1084ff",
                "borderRadius": 20,
                "marginTop": 5,
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 32,
                  "paddingTop": 5,
                }
              }
            >
              my-text
            </Text>
            <Text
              style={
                Object {
                  "alignContent": "flex-end",
                  "color": "white",
                  "paddingTop": 5,
                }
              }
            >
              my-author
               from 
              my-country
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Share awesomeness
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders no connection notify if disconnected from internet 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#89cff0",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
  >
    <Image
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 64,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="headerContainer"
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        accessibilityTraits="header"
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontSize": 17,
            "fontWeight": "600",
            "marginHorizontal": 16,
          }
        }
      >
        Home
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#b52424",
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      No Internet Connection
    </Text>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#89cff0",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "flex": 1,
        }
      }
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
        />
      }
    >
      <RCTRefreshControl />
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#1084ff",
                "borderRadius": 20,
                "marginTop": 5,
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 32,
                  "paddingTop": 5,
                }
              }
            >
              Loading...
            </Text>
            <Text
              style={
                Object {
                  "alignContent": "flex-end",
                  "color": "white",
                  "paddingTop": 5,
                }
              }
            >
              
               from 
              
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Share awesomeness
        </Text>
      </View>
    </View>
  </View>
</View>
`;
